sparse/unit_test/Test_Sparse_sptrsv.hpp

```diff
@@ -38,1111 +38,557 @@ using namespace KokkosKernels;
 
       sum = 0.0;
-      Kokkos::parallel_reduce(
-          Kokkos::RangePolicy<typename device::execution_space>(0,
-                                                                lhs.extent(0)),
-          ReductionCheck<ValuesType, scalar_t, lno_t>(lhs), sum);
-      if (sum != lhs.extent(0)) {
-        std::cout << "Upper Tri Solve FAILURE" << std::endl;
-        kh.get_sptrsv_handle()->print_algorithm();
+      Kokkos::parallel_reduce(range_policy_t(0, lhs.extent(0)), KOKKOS_LAMBDA ( const lno_t i, scalar_t &tsum ) {
+          tsum += lhs(i);
+        }, sum);
+      if ( sum != lhs.extent(0) ) {
+        std::cout << "Lower Tri Solve FAILURE" << std::endl;
+      }
+      else {
+        std::cout << "Lower Tri Solve SUCCESS!" << std::endl;
+        //std::cout << "Num-levels = " << kh->get_sptrsv_handle()->get_num_levels() << std::endl;
       }
-      EXPECT_TRUE(sum == scalar_t(lhs.extent(0)));
+      EXPECT_TRUE( sum == scalar_t(lhs.extent(0)) );
 
-      // FIXME Issues with various integral type combos - algorithm currently
-      // unavailable and commented out until fixed
-      /*
-      Kokkos::deep_copy(lhs, ZERO);
+      Kokkos::deep_copy(lhs, 0);
       kh.get_sptrsv_handle()->set_algorithm(SPTRSVAlgorithm::SEQLVLSCHED_TP2);
+      kh.get_sptrsv_handle()->print_algorithm();
+      timer.reset();
       sptrsv_solve( &kh, row_map, entries, values, rhs, lhs );
-      Kokkos::fence();
+      std::cout << "LTRI Solve SEQLVLSCHED_TP2 Time: " << timer.seconds() << std::endl;
 
       sum = 0.0;
-      Kokkos::parallel_reduce( Kokkos::RangePolicy<typename
-      device::execution_space>(0, lhs.extent(0)), ReductionCheck<ValuesType,
-      scalar_t, lno_t>(lhs), sum); if ( sum != lhs.extent(0) ) { std::cout <<
-      "Upper Tri Solve FAILURE" << std::endl;
-        kh.get_sptrsv_handle()->print_algorithm();
+      Kokkos::parallel_reduce(range_policy_t(0, lhs.extent(0)), KOKKOS_LAMBDA ( const lno_t i, scalar_t &tsum ) {
+          tsum += lhs(i);
+        }, sum);
+      if ( sum != lhs.extent(0) ) {
+        std::cout << "Lower Tri Solve FAILURE" << std::endl;
+      }
+      else {
+        std::cout << "Lower Tri Solve SUCCESS!" << std::endl;
+        //std::cout << "Num-levels = " << kh->get_sptrsv_handle()->get_num_levels() << std::endl;
       }
       EXPECT_TRUE( sum == scalar_t(lhs.extent(0)) );
-      */
 
       kh.destroy_sptrsv_handle();
     }
-
+    // Upper tri
+    std::cout << "UpperTriTest Begin" << std::endl;
     {
-      Kokkos::deep_copy(lhs, ZERO);
-      KernelHandle kh;
-      bool is_lower_tri = false;
-      kh.create_sptrsv_handle(SPTRSVAlgorithm::SEQLVLSCHD_TP1CHAIN, nrows,
-                              is_lower_tri);
-      auto chain_threshold = 1;
-      kh.get_sptrsv_handle()->reset_chain_threshold(chain_threshold);
+      std::string mtx_filename = "/ascldap/users/ndellin/TestCodes-GitlabEx/KokkosEcoCodes/KokkosKernels-DevTests/Matrices/U-dielFilterV2real-amd.mtx";
```

Comment:
I recommend removing the hard coded directory structure too

Reply:
@srajama1 , this entire test is `#if 0` 'd out. I cleaned it up a little just in case we wanted it in the future, but I'd also be happy to just delete it. I don't know why it was kept despite being inactive. @vqd8a , do you know?

Reply:
@jgfouca I do not know. I think it is in there for a long time, probably when @ndellingwood developed the sptrsv. We can delete it.

Reply:
@vqd8a , OK, I deleted it.